pyipmi/fields.py

```diff
@@ -38,10 +39,10 @@ def _decode_data(self, data):
         """`data` is array.array"""
         self.major = data[0]
 
-        if data[1] is 0xff:
+        if data[1] == 0xff:
             self.minor = data[1]
         elif data[1] <= 0x99:
-            self.minor = int(data[1:2].tostring().decode('bcd+'))
+            self.minor = int(py3_array_tobytes(data[1:2]).decode('bcd+'))
```

Comment:
It can be written as
```suggestion
            self.minor = int(bchr(data[1]).decode('bcd+'))
```
where `bchr(x)` is `chr(x)` in Python 2 and `bytes([x])` in Python 3.